docs/docs-graphql/schema.graphqls

```diff
@@ -1630,7 +1630,10 @@ type GitRef {
   id: GitRefID!
 
   """The filesystem tree at this ref."""
-  tree: Directory!
+  tree(
+    """Set to true to discard .git directory."""
+    discardGitDir: Boolean = false
```

Comment:
Mini-bikeshed: I wonder if we even need this? With #8436 you could just exclude `.git` at the argument site, which would then be applied no matter how the `Directory` was created. But - I guess you don't want every call site to have to anticipate and exclude a `.git` directory. So it kind of makes sense to do it before passing it in, too. But you could also do `.withoutDirectory(".git")` which might be even less typing depending on your SDK. 🤷‍♂️

Reply:
There's an issue with Python uv that hold #8436 so for now we cannot merge it, but yeah I think it's not a good DX to force `// +ignore=[".git/"]` on every call.

Reply:
Good point. I'm happy with the idea of removing it, but also not opposed to keeping it.

Reply:
I think the point that this would prevent you from needing to always try to ignore/remove the dir out of caution is enough justification for keeping this option in place.